src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -94,9 +94,9 @@ public function testSpecifications()
                 }
 
                 $test = $this->parser->parse($yaml);
-                if (isset($test['dump_skip']) && $test['dump_skip']) {
+                if (!empty($test['dump_skip'])) {
```

Comment:
Please remove this commit switching to `!empty`, as PHP's `empty` is a bit weird, which means your new code is not actually equivalent: https://3v4l.org/komCe

Reply:
You forgot a negation, it has to be `!empty($a)`.
Here is a more complete example, you can add more values, and try to come up with one that fails: https://3v4l.org/91tGD
I have yet to see a case where this fails.

Reply:
> but '0' is truthy, but it is not "not empty".
Nope.
Here is your corrected example: https://3v4l.org/ehSH1
`'0'` is empty and falsy = not truthy.

> be careful that truthiness checks (in conditions) is not the same than casting to a boolean

I have yet to see an example where this is not the case.
See the updated https://3v4l.org/cYEfG
I see only two groups: One for values that are empty and falsy, another for values that are not empty and truthy.

Reply:
From https://www.php.net/manual/en/function.empty.php

> That means empty() is essentially the concise equivalent to !isset($var) || $var == false.

From https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.types

> | Type of Operand 1 | Type of Operand 2 | Result |
> | ----------------- | ------------ | ----- |
> | bool or null | anything | Convert both sides to bool, false < true |

Reply:
As a general code-style rule, we try to not use `empty()` unless the alternative is worse.
The reason is this exact discussion: we don't want ppl to have to precisely know its rules to be able to understand the code. The previous code was doing the same isn't it? Then our CS would be to keep it as is.
Alternatively, we might want to go with `if ($test['dump_skip'] ?? false) {`, but I don't see any hurry to do so.

Reply:
I am ok to drop/revert the change (or use `??`), _but_ I claim that I am still factually correct in my previous comments.
We all like to be factually correct on the internet. And we can all learn a bit.

> Alternatively, we might want to go with `if ($test['dump_skip'] ?? false) {`,

Yes!
I wanted to do this while the PR was for 6.1. See fdf21ce1986e8de2d68bae8a5a25efb1a2fc8897 in #46731. Then I thought we have to drop it for PHP 7.1. But actually we can still use it, since 7.0! https://3v4l.org/7llO8 (eol versions)

> but I don't see any hurry to do so.

None of these "refactoring" changes are super urgent, I would say. But I think it is not wrong to do it in one go.

(rant on)

> we don't want ppl to have to precisely know its rules to be able to understand the code.

Reminds me of Friday 13, or GMO food.
But this is fine. We _do_ omit floor 13 or door 13 in some buildings, because the customer is king.
But really, the function is not mysterious, and the documentation is very explicit about the equivalence to bool cast.

Also, we _do_ need to understand how `empty()` works, because this is also the default "callback" in `array_filter()`.
And we already need to learn about the rules of cast to bool.

There are other concerns I would have about both `isset()` and `empty()`:
- Both of them obfuscate undefined or misspelled variables (but the IDE will tell us, hopefully).
- `isset()` only works on variables, not on non-variable expressions. E.g. `isset(5)` is not allowed. For `empty()` this is fine.
- You could argue that `empty()` does more than one thing (= one thing too many).
- Both are language constructs, not regular functions, so we cannot use them with `call_user_func()` or similar.
- You could argue that it is not well enough documented on php.net... but hey, it actually is! The page really says it all.

Actually `if ($x ?? false)` still has some of these flaws.

(rant off)